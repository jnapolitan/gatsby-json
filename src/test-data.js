module.exports = {
  activities: [{
      "type": "Text",
      "content": "Mindfulness is a type of meditation. It can be practiced both formally (i.e. sitting down and focusing on the breath or body intentionally for a few minutes) or informally (i.e. bringing your focus and attention to an activity to the moment while you engage in an activity).",
      "prompt": null,
      "options": null
    },
    {
      "type": "Text",
      "content": "After only eight minutes of mindfulness practice people often start to notice a calmer mind.",
      "prompt": null,
      "options": null
    },
    {
      "type": "Text",
      "content": "After two weeks people report better focus and memory. Overall, most people will start to notice changes in a few weeks when they practice for 10-20 minutes a day but keep in mind that this will be different for everyone. Pay attention to your own experience - mindfulness can help you develop more patience with this too.",
      "prompt": null,
      "options": null
    },
    {
      "type": "Question",
      "content": null,
      "prompt": "How do you want to use mindfulness to improve your work?",
      "options": [
        "Increase Focus",
        "Improve Concentration",
        "Mental Clarity",
        "Reduce Stress",
        "Respond With Kindness",
        "More Self-Control",
        "Other"
      ]
    }
  ],
  section: {
    "programId": "1",
    "programName": "core-pillars-program",
    "order": 1,
    "name": "Mindfulness",
    "image": "https://my.joinmodernhealth.com/static/media/meditation-image-page.71f7c8b3.png",
    "activities": [{
        "type": "Text"
      },
      {
        "type": "Text"
      },
      {
        "type": "Text"
      },
      {
        "type": "Question"
      }
    ]
  },
  program: {
    "node": {
      "id": "1",
      "name": "Core Pillars Program",
      "description": "This program is for people who want to live a happier and more fulfilling life. It's for those of us who want to become more mindful, more engaged, and more connected to our everyday lives and relationships.",
      "sections": [{
          "programId": "1",
          "programName": "core-pillars-program",
          "order": 1,
          "name": "Mindfulness",
          "image": "https://my.joinmodernhealth.com/static/media/meditation-image-page.71f7c8b3.png",
          "activities": [{
              "type": "Text"
            },
            {
              "type": "Text"
            },
            {
              "type": "Text"
            },
            {
              "type": "Question"
            }
          ]
        },
        {
          "programId": "1",
          "programName": "core-pillars-program",
          "order": 2,
          "name": "Values",
          "image": "https://my.joinmodernhealth.com/static/media/values.e8b1dc68.png",
          "activities": [{
              "type": "Text"
            },
            {
              "type": "Text"
            },
            {
              "type": "Text"
            },
            {
              "type": "Text"
            },
            {
              "type": "Question"
            }
          ]
        },
        {
          "programId": "1",
          "programName": "core-pillars-program",
          "order": 3,
          "name": "Action",
          "image": "https://my.joinmodernhealth.com/static/media/action.605a754d.png",
          "activities": [{
              "type": "Text"
            },
            {
              "type": "Text"
            },
            {
              "type": "Text"
            }
          ]
        },
        {
          "programId": "1",
          "programName": "core-pillars-program",
          "order": 4,
          "name": "Flexibility",
          "image": "https://my.joinmodernhealth.com/static/media/flexibility.edd8dc40.png",
          "activities": [{
              "type": "Text"
            },
            {
              "type": "Text"
            },
            {
              "type": "Text"
            },
            {
              "type": "Question"
            },
            {
              "type": "Text"
            }
          ]
        }
      ]
    }
  },
  questionActivity: {
    "type": "Question",
    "content": null,
    "prompt": "How do you want to use mindfulness to improve your work?",
    "options": [
      "Increase Focus",
      "Improve Concentration",
      "Mental Clarity",
      "Reduce Stress",
      "Respond With Kindness",
      "More Self-Control",
      "Other"
    ]
  },
  textActivity: {
    "type": "Text",
    "content": "After two weeks people report better focus and memory. Overall, most people will start to notice changes in a few weeks when they practice for 10-20 minutes a day but keep in mind that this will be different for everyone. Pay attention to your own experience - mindfulness can help you develop more patience with this too.",
    "prompt": null,
    "options": null
  },
  indexData: {
    "allProgramsJson": {
      "edges": [{
          "node": {
            "id": "1",
            "name": "Core Pillars Program",
            "description": "This program is for people who want to live a happier and more fulfilling life. It's for those of us who want to become more mindful, more engaged, and more connected to our everyday lives and relationships.",
            "sections": [{
                "programId": "1",
                "programName": "core-pillars-program",
                "order": 1,
                "name": "Mindfulness",
                "image": "https://my.joinmodernhealth.com/static/media/meditation-image-page.71f7c8b3.png",
                "activities": [{
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  },
                  {
                    "type": "Question"
                  }
                ]
              },
              {
                "programId": "1",
                "programName": "core-pillars-program",
                "order": 2,
                "name": "Values",
                "image": "https://my.joinmodernhealth.com/static/media/values.e8b1dc68.png",
                "activities": [{
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  },
                  {
                    "type": "Question"
                  }
                ]
              },
              {
                "programId": "1",
                "programName": "core-pillars-program",
                "order": 3,
                "name": "Action",
                "image": "https://my.joinmodernhealth.com/static/media/action.605a754d.png",
                "activities": [{
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  }
                ]
              },
              {
                "programId": "1",
                "programName": "core-pillars-program",
                "order": 4,
                "name": "Flexibility",
                "image": "https://my.joinmodernhealth.com/static/media/flexibility.edd8dc40.png",
                "activities": [{
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  },
                  {
                    "type": "Question"
                  },
                  {
                    "type": "Text"
                  }
                ]
              }
            ]
          }
        },
        {
          "node": {
            "id": "2",
            "name": "Psychological First Aid",
            "description": "Experiencing distress after a stressful or traumatic event is common. The goals of this short course are to help you cope and recover.",
            "sections": [{
                "programId": "2",
                "programName": "psychological-first-aid",
                "order": 1,
                "name": "Common Reactions you Might Experience",
                "image": "https://staging.joinmodernhealth.com/static/media/psychological-first-aid.38f004a7.svg",
                "activities": [{
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  }
                ]
              },
              {
                "programId": "2",
                "programName": "psychological-first-aid",
                "order": 2,
                "name": "What Helps (and what doesn't)",
                "image": "https://staging.joinmodernhealth.com/static/media/undraw-feeling-blue-4-b-7-q-copy.74c19b42.svg",
                "activities": [{
                    "type": "Text"
                  },
                  {
                    "type": "Text"
                  }
                ]
              }
            ]
          }
        }
      ]
    }
  },
  sectionData: {
    "section": {
      "programId": "1",
      "programName": "core-pillars-program",
      "name": "Mindfulness",
      "description": "Mindfulness is the ability to be fully present and aware of in the current moment, without judging your experience or yourself.",
      "image": "https://my.joinmodernhealth.com/static/media/meditation-image-page.71f7c8b3.png",
      "order": 1,
      "endOrder": 4,
      "activities": [{
          "type": "Text",
          "content": "Mindfulness is a type of meditation. It can be practiced both formally (i.e. sitting down and focusing on the breath or body intentionally for a few minutes) or informally (i.e. bringing your focus and attention to an activity to the moment while you engage in an activity).",
          "prompt": null,
          "options": null
        },
        {
          "type": "Text",
          "content": "After only eight minutes of mindfulness practice people often start to notice a calmer mind.",
          "prompt": null,
          "options": null
        },
        {
          "type": "Text",
          "content": "After two weeks people report better focus and memory. Overall, most people will start to notice changes in a few weeks when they practice for 10-20 minutes a day but keep in mind that this will be different for everyone. Pay attention to your own experience - mindfulness can help you develop more patience with this too.",
          "prompt": null,
          "options": null
        },
        {
          "type": "Question",
          "content": null,
          "prompt": "How do you want to use mindfulness to improve your work?",
          "options": [
            "Increase Focus",
            "Improve Concentration",
            "Mental Clarity",
            "Reduce Stress",
            "Respond With Kindness",
            "More Self-Control",
            "Other"
          ]
        }
      ]
    }
  }
};