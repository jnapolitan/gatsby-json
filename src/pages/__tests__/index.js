import React from "react";
import renderer from "react-test-renderer";
import Index from "../index";
import { indexData } from "../../test-data";

describe("Index", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Index data={indexData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});