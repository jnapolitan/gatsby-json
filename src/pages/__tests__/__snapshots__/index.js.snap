// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
Array [
  <header>
    <div
      className="container"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          className="headerImg"
          src="test-file-stub"
        />
      </a>
    </div>
  </header>,
  <div
    className="container"
  >
    <main>
      <div
        className="header"
      >
        <img
          alt="hand-with-heart"
          src="test-file-stub"
        />
        <div>
          <h2>
            Program Overview
          </h2>
          <p>
            Welcome to your Program Overview page. Here you'll find helpful resources and activities to help you manage your mental well-being so you can stay happy and healthy.
          </p>
        </div>
      </div>
      <div
        className="container"
      >
        <div
          className="container"
        >
          <h3>
            Core Pillars Program
          </h3>
          <button
            className="modalButton"
            onClick={[Function]}
          >
            Learn more
          </button>
        </div>
        <div
          className="sections"
        >
          <a
            href="/core-pillars-program/part-1"
          >
            <div
              className="container"
            >
              <img
                alt="Mindfulness"
                src="https://my.joinmodernhealth.com/static/media/meditation-image-page.71f7c8b3.png"
              />
              <div>
                <p
                  className="partText"
                >
                  Part 
                  One
                </p>
                <h4>
                  Mindfulness
                </h4>
              </div>
            </div>
          </a>
          <a
            href="/core-pillars-program/part-2"
          >
            <div
              className="container"
            >
              <img
                alt="Values"
                src="https://my.joinmodernhealth.com/static/media/values.e8b1dc68.png"
              />
              <div>
                <p
                  className="partText"
                >
                  Part 
                  Two
                </p>
                <h4>
                  Values
                </h4>
              </div>
            </div>
          </a>
          <a
            href="/core-pillars-program/part-3"
          >
            <div
              className="container"
            >
              <img
                alt="Action"
                src="https://my.joinmodernhealth.com/static/media/action.605a754d.png"
              />
              <div>
                <p
                  className="partText"
                >
                  Part 
                  Three
                </p>
                <h4>
                  Action
                </h4>
              </div>
            </div>
          </a>
          <a
            href="/core-pillars-program/part-4"
          >
            <div
              className="container"
            >
              <img
                alt="Flexibility"
                src="https://my.joinmodernhealth.com/static/media/flexibility.edd8dc40.png"
              />
              <div>
                <p
                  className="partText"
                >
                  Part 
                  Four
                </p>
                <h4>
                  Flexibility
                </h4>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        className="container"
      >
        <div
          className="container"
        >
          <h3>
            Psychological First Aid
          </h3>
          <button
            className="modalButton"
            onClick={[Function]}
          >
            Learn more
          </button>
        </div>
        <div
          className="sections"
        >
          <a
            href="/psychological-first-aid/part-1"
          >
            <div
              className="container"
            >
              <img
                alt="Common Reactions you Might Experience"
                src="https://staging.joinmodernhealth.com/static/media/psychological-first-aid.38f004a7.svg"
              />
              <div>
                <p
                  className="partText"
                >
                  Part 
                  One
                </p>
                <h4>
                  Common Reactions you Might Experience
                </h4>
              </div>
            </div>
          </a>
          <a
            href="/psychological-first-aid/part-2"
          >
            <div
              className="container"
            >
              <img
                alt="What Helps (and what doesn't)"
                src="https://staging.joinmodernhealth.com/static/media/undraw-feeling-blue-4-b-7-q-copy.74c19b42.svg"
              />
              <div>
                <p
                  className="partText"
                >
                  Part 
                  Two
                </p>
                <h4>
                  What Helps (and what doesn't)
                </h4>
              </div>
            </div>
          </a>
        </div>
      </div>
    </main>
  </div>,
]
`;
